yt_dlp/extractor/beeg.py

```diff
@@ -36,78 +50,54 @@ class BeegIE(InfoExtractor):
         # api/v6 v2 w/o t
         'url': 'https://beeg.com/1277207756',
         'only_matching': True,
-    }, {
-        'url': 'https://beeg.porn/video/5416503',
-        'only_matching': True,
-    }, {
-        'url': 'https://beeg.porn/5416503',
-        'only_matching': True,
     }]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
         webpage = self._download_webpage(url, video_id)
 
-        beeg_version = self._search_regex(
-            r'beeg_version\s*=\s*([\da-zA-Z_-]+)', webpage, 'beeg version',
-            default='1546225636701')
-
-        if len(video_id) >= 10:
-            query = {
-                'v': 2,
-            }
-            qs = parse_qs(url)
-            t = qs.get('t', [''])[0].split('-')
-            if len(t) > 1:
-                query.update({
-                    's': t[0],
-                    'e': t[1],
-                })
-        else:
-            query = {'v': 1}
+        video = self._download_json(
+            'https://store.externulls.com/facts/file/%s' % video_id,
+            video_id, 'Downloading JSON for %s' % video_id)
 
-        for api_path in ('', 'api.'):
-            video = self._download_json(
-                'https://%sbeeg.com/api/v6/%s/video/%s'
-                % (api_path, beeg_version, video_id), video_id,
-                fatal=api_path == 'api.', query=query)
-            if video:
-                break
+        resources = video['file']['hls_resources'] if traverse_obj(video, ('file', 'hls_resources')) else video['fc_facts'][0]['hls_resources']
 
         formats = []
-        for format_id, video_url in video.items():
-            if not video_url:
+        for format_id, video_uri in resources.items():
+            if not video_uri:
                 continue
             height = self._search_regex(
-                r'^(\d+)[pP]$', format_id, 'height', default=None)
+                r'fl_cdn_(\d+)', format_id, 'height', default=None)
             if not height:
                 continue
+#                formats.extend(self._extract_m3u8_formats(
+#                'https://video.beeg.com/' + video_uri, video_id, ext='m3u8', m3u8_id='hls'))
```

Comment:
What happens when you use this code?

Reply:
wow, it works, i don't know what i did wrong yesterday.

but, it doesn't return the best video + audio.
listing formats we can see the difference.
with extract_m3u8_formats
![image](https://user-images.githubusercontent.com/216170/152366065-4ddef09f-e1cd-4ce1-9e76-2572c1e73464.png)
generated video.
![image](https://user-images.githubusercontent.com/216170/152366167-38756e79-afd4-43cc-a528-b1c16e50a7a5.png)

Manual extract.
![image](https://user-images.githubusercontent.com/216170/152366457-56953794-a4cb-4581-a1c9-006ba816f723.png)
generated video
![image](https://user-images.githubusercontent.com/216170/152366642-4914fd78-0db5-4e0a-873b-e1cd028380bc.png)

Reply:
You can manually fill in the `height`/`width` in the formats that extract_m3u8 returns